subsys/net/ip/l2/ieee802154/ieee802154_radio_aloha.c

```diff
@@ -39,7 +39,7 @@ static inline int aloha_tx_fragment(struct net_if *iface,
 			continue;
 		}
 
-		ret = wait_for_ack(ctx, ack_required);
+		ret = wait_for_ack(iface, ack_required);
```

Comment:
I started to play with it and noticed it is not enough. The driver must know whether the hardware should wait for an ACK or not. Not every packet may claim an ACK. Since it is not desired that the driver parses the FCS field, the TX function need to be changed like 
`int (*tx)(struct device*dev, struct net_pkt *pkt, struct net_buf *frag, bool ack_required); `

Reply:
indeed, your mrc20a needs to know that it if I remember well.

That's where chip manufacturers are not consistent: they handle AR flag properly on rx side to quickly respond with an ack frame, which mean their hardware (or inner firmware) is clever enough to parse the FCF to do so. But on tx? nope, letting the driver to do all the heavy lifting.

I think I prefer that drivers do a fcf parse by themselves. I'll add a quick utility function for this (after all it's just a matter of checking one bit in the very first 2 bytes of the ll reserve).

Reply:
There is ieee802154_ack_required(...) in ieee802154_frame.h that may be used for it, but it is not exposed to public.

Reply:
Yes but it would mean exposing also the frame structures and all, and that I don't want to.

I'll make a simpler one.

Reply:
@tbursztyka : @dleach02 raised the same issue in a separate discussion: that a driver may need to look into a packet, and those are defined in subsystem/net/ip/l2/ieee802154/ , instead of include/ . Can you elaborate what's the problem with moving those defines to public includes? We have L3 headers defined publicly, what's the problem with doing the same for standardized L2 technologies?

@dleach02 , I'd let you continue from here, to not serve as a "broken phone" proxy ;-).

Reply:
I also ran into the "driver needs to know about ACK request" and was going to just look at the AR bit in the FC. But this brings up another problem with our include files because the information on 802.15.4 header structure are in header files in the subsystem and not in the /include area (Linux has it up there). But I also think we may want to consider a transmit API that includes control information on the transmit where one item in this control information is ACK offload requirement.

Reply:
I don't think it is bad to expose IEEE802.15.4 frame structures in a more global include that is accessible by the interface drivers. It isn't internal stuff that would be proprietary to the Zephyr 15.4 stack. 

But I realized that I didn't need it yet since the current state of the system is that there isn't ACK offload (except for what is being done here).

I do think there should be a larger discussion about the PLME interface being exposed by the PHY drivers and what future features we want/need to support. The NXP HW has a number of interesting features that we may want to expose through the interface depending on how we wanted the Zephyr stack to handle things (note that I'm new to the NXP HW as well even though I'm an NXP person). For example, when we get around to supporting modes where we want to schedule RX/TX times we would likely want to enable/disable the RX based on this information to conserve power. From what I can see now, the 15.4 stack expects the RX to be enabled all the time, which is likely good enough for now but as we optimize things in the future we may want to consider how to abstract something like this and when to leverage the underlying HW to do some of this.